### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by Koliras**

### CODE REVIEW

**Review by davinnchii — COMMENTED** [review]
Well done =)
I'd prefer to use forEach instead of common for-loop. Also in my opinion, adding a empty line before `break` will improve readability of your code.

**Review by davinnchii — APPROVED** [review]

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,31 @@
...
-  // write code here
+  const stateCopy = { ...state };
+  const stateVersions = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+      case 'removeProperties':
```
Comment (Moroz-Dmytro) [inline src/transformStateWithClones.js:18]:
```suggestion
        break;
        
      case 'removeProperties':
```

Comment (Moroz-Dmytro) [reply src/transformStateWithClones.js:18]: And in other places, will be easier to read

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,31 @@
...
+        action.keysToRemove.forEach(key => {
+          delete stateCopy[key];
+        });
+        break;
+      case 'clear':
+        Object.keys(stateCopy).forEach(key => {
+          delete stateCopy[key];
+        });
+        break;
+      default: break;
```
Comment (Moroz-Dmytro) [inline src/transformStateWithClones.js:28]: Would be nice to throw an error in such case

**Review by Moroz-Dmytro — CHANGES_REQUESTED** [review]
Good job 👍 
Small comments

**Review by tiserett — CHANGES_REQUESTED** [review]

**Pull request closed without merging**